### ISSUE TITLE
Store failure reason in metadata

### ISSUE THREAD

**Issue by twaugh** [projectatomic/atomic-reactor#425]
When a build fails, `store_metadata_in_osv3` should store the reason for that failure.

Eg plugin name during which the failure occurred, or whether it was during docker build. Maybe also an error name or description.

**Comment by TomasTomecek** [comment]
This would improve user experience, definitely. Usually it's pretty obvious for us, developers of the tool, who saw them logs thousands of times, but not so obvious for the rest of the world. So +1 on this.

**Comment by twaugh** [comment]
Yes. So in addition to putting the plugin name (e.g. squash) and error name (e.g. SquashFailure), we should also make the final traceback output in the log much better. It isn't clear that this is a repeat of an earlier traceback.
